<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Document</title>
</head>
<body>
  <script>
    // setTimeout(() => {
    //   console.log('hello world!')
    // }, 1000)

    // promise参数：函数(resolve，reject)
    // resolve, reject它们本身又是函数
    // 链式编程
    // new Promise((resolve, reject) => {
    //   // 第一次网络请求的代码
    //   setTimeout(() => {
    //     resolve()
    //   }, 1000)
    // }).then(() => {
    //   // 第一次拿到结果的处理代码
    //   console.log('hello world!')
    //   console.log('hello world!')
    //   console.log('hello world!')
    //   console.log('hello world!')
    //   console.log('hello world!')

    //   return new Promise((resolve, reject) => {
    //     // 第二次网络请求的代码
    //     setTimeout(() => {
    //       resolve()
    //     }, 1000)
    //   }).then(() => {
    //     // 第二次拿到结果的处理代码
    //     console.log('hello vue!')
    //     console.log('hello vue!')
    //     console.log('hello vue!')
    //     console.log('hello vue!')

    //     return new Promise((resolve, reject) => {
    //       // 第三次网络请求的代码
    //       setTimeout(() => {
    //         resolve()
    //       }, 1000)
    //     }).then(() => {
    //       // 第三次拿到结果的处理代码
    //       console.log('hello python!')
    //       console.log('hello python!')
    //       console.log('hello python!')
    //     })
    //   })
    // })

    // 什么情况下会用到promise？
    // --一般情况下，是有异步操作时用promise进行封装
    // new -> 构造函数（1.保存了一些状态信息 2.执行传入的函数）
    // 在执行传入的回调函数时，会传入两个参数，resolve,reject，本身又是函数
    new Promise((resolve, reject) => {
      setTimeout(() => {
        // 成功时调用resolve
        // resolve('hello world')
        // 失败时调用reject
        reject('error message')
      }, 1000)
    }).then((data) => {
      console.log(data);
      console.log(data);
      console.log(data);
      console.log(data);
    }).catch((err) => {
      console.log(err);
    })
  </script>
</body>
</html>